mesonbuild/dependencies/cmake.py

```diff
@@ -573,7 +574,7 @@ def _setup_cmake_dir(self, cmake_file: str) -> Path:
         cmake_cache = build_dir / 'CMakeCache.txt'
         cmake_files = build_dir / 'CMakeFiles'
         if cmake_cache.exists():
-            cmake_cache.unlink()
+            windows_proof_rm(cmake_cache.absolute().as_posix())
```

Comment:
In scipy/scipy#19855 and other recent PRs, I've noticed that our Windows CI builds will sporadically fail. I think the issue stems from this particular delete. While I was changing this one, I decided I might as well change all of them. I don't know if that was overkill.

Reply:
Can you show me an example log?

Reply:
Here is the relevant excerpt from such a log:
```
SciPy 1.13.0.dev0

  User defined options
    Native files: D:\a\scipy\scipy\.mesonpy-rct2q19y\meson-python-native-file.ini
    buildtype   : release
    b_ndebug    : if-release
    b_vscrt     : md
    use-pythran : false

Found ninja.EXE-1.11.1.git.kitware.jobserver-1 at C:\hostedtoolcache\windows\Python\3.11.7\x64\Scripts\ninja.EXE
+ meson dist --allow-dirty --no-tests --formats gztar
WARNING: Repository has uncommitted changes that will not be included in the dist tarball
Created D:\a\scipy\scipy\.mesonpy-rct2q19y\meson-dist\SciPy-1.13.0.dev0.tar.gz
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\shutil.py", line 624, in _rmtree_unsafe
    os.rmdir(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: '.\\.mesonpy-rct2q19y\\meson-private\\cmake_scipy-openblas\\CMakeFiles\\CMakeScratch\\TryCompile-cewmca'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\tempfile.py", line 878, in onerror
    _os.unlink(path)
PermissionError: [WinError 5] Access is denied: '.\\.mesonpy-rct2q19y\\meson-private\\cmake_scipy-openblas\\CMakeFiles\\CMakeScratch\\TryCompile-cewmca'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\shutil.py", line 624, in _rmtree_unsafe
    os.rmdir(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: '.\\.mesonpy-rct2q19y\\meson-private\\cmake_scipy-openblas\\CMakeFiles\\CMakeScratch\\TryCompile-cewmca'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\tempfile.py", line 878, in onerror
    _os.unlink(path)
PermissionError: [WinError 5] Access is denied: '.\\.mesonpy-rct2q19y\\meson-private\\cmake_scipy-openblas\\CMakeFiles\\CMakeScratch\\TryCompile-cewmca'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\shutil.py", line 624, in _rmtree_unsafe
    os.rmdir(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: '.\\.mesonpy-rct2q19y\\meson-private\\cmake_scipy-openblas\\CMakeFiles\\CMakeScratch\\TryCompile-cewmca'
```
It proceeds to alternate between these two functions thousands of times until reaching a recursion depth error.

Reply:
Sorry I commented under the wrong line. The problematic line should be one below this one.